# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../compositeproto/long64.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

On at least powerpc64 long is 64bit, and since XID, used for Window and Pixmap
are 64bit then some protocol fields are not properly defined.

I tried serveral attempts to fix this bug, including fixing the main X headers
however it turned out it all has too many sideeffects. See X.org bug #1425 for
a complete trace with additional comments.

On x86-64 libXcomposite only works accidently due the endianess differnce, the
structure is miss-sized there as well.

  https://bugs.freedesktop.org/show_bug.cgi?id=1425

 - Rene Rebe <rene@exactcode.de>

--- compositeproto-0.2.2/compositeproto.h.orig	2006-02-07 12:35:03.000000000 +0100
+++ compositeproto-0.2.2/compositeproto.h	2006-02-07 12:35:17.000000000 +0100
@@ -30,6 +30,7 @@
 
 #define Window CARD32
 #define Region CARD32
+#define Pixmap CARD32
 
 /* 
  * requests and replies
